### PR TITLE
Gui refactored!

### PR DESCRIPTION
**Pull request by blazeuser**
Refactoring of screen package. A container class has been created to hold all services. This has been done in order to get rid of static method calls in other classes. This makes testing easier since you can just mock that container class and its dependencies. This PR has no new tests because it hasn't been confirmed they all run together since they require a JavaFX thread.

### CODE REVIEW

File: src/main/java/logging/LoggerFactory.java [hunk 1]
```diff
@@ -17,25 +18,35 @@
...
-   public LoggerFactory(FileSystem fs) {
+   private LoggerFactory(FileSystem fs) {
         fileSystem = fs;
         fileSystem.clearFile();
-        logger = new Logger(this.getClass(), fileSystem);
+        logger = new Logger(this.getClass(), fileSystem, serviceLocator);
         logger.info("a new logger factory has been created");
    }
 
+   public static void register(ServiceLocator sL) {
```
Comment (tvheems) [inline src/main/java/logging/LoggerFactory.java]: Missing javadoc.

File: src/main/java/filesystem/FileSystem.java [hunk 2]
```diff
@@ -37,8 +39,13 @@ public FileSystem() {
         } catch (IOException e) {
             e.printStackTrace();
         }
+    }
 
-
+    public static void register(ServiceLocator sL) {
```
Comment (tvheems) [inline src/main/java/filesystem/FileSystem.java]: Missing javadoc

File: src/main/java/screens/Backlog.java [hunk 3]
```diff
@@ -36,6 +39,14 @@
         this.setScene(scene);
     }
 
+    public static void register(ServiceLocator sL) {
```
Comment (tvheems) [inline src/main/java/screens/Backlog.java]: Missing javadoc.

File: src/main/java/screens/FXElementsFactory.java [hunk 4]
```diff
@@ -4,12 +4,24 @@
...
 /**
  * Factory for JavaFX constructors.
  */
 public class FXElementsFactory {
 
+    private FXElementsFactory() {
+
+    }
+
+    public static void register(ServiceLocator sL) {
```
Comment (tvheems) [inline src/main/java/screens/FXElementsFactory.java]: Add javadoc to all "register" methods please.

File: src/main/java/filesystem/FileSystem.java (hunk 2 above)
Comment (MStraatman) [inline src/main/java/filesystem/FileSystem.java]: Please add some JavaDoc to this method.

File: src/main/java/screens/nodehandlers/NodeCenter.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleNode(DrawNode node) {
+        serviceLocator.getGraphScene().drawGraph(node.getIndex(), serviceLocator.getController().getRadius());
+        serviceLocator.getGraphScene().switchToInfo();
+    }
+
+    @Override
+    public void handleLine(Line line) {
```
Comment (tvheems) [inline src/main/java/screens/nodehandlers/NodeCenter.java]: Why is this empty?

File: src/main/java/logging/LoggerFactory.java [hunk 6]
```diff
@@ -17,25 +18,35 @@
      */
     private final FileSystem fileSystem;
 
+    private static ServiceLocator serviceLocator;
```
Comment (MStraatman) [inline src/main/java/logging/LoggerFactory.java]: Javadoc.

File: src/main/java/logging/LoggerFactory.java (hunk 1 above)
Comment (MStraatman) [inline src/main/java/logging/LoggerFactory.java]: Javadoc.

File: src/main/java/screens/Backlog.java [hunk 7]
```diff
@@ -14,6 +15,8 @@
      */
     private final TextArea textArea;
 
+    private static ServiceLocator serviceLocator;
```
Comment (MStraatman) [inline src/main/java/screens/Backlog.java]: Javadoc.

File: src/main/java/screens/Backlog.java (hunk 3 above)
Comment (MStraatman) [inline src/main/java/screens/Backlog.java]: Javadoc.

File: src/main/java/screens/FXElementsFactory.java [hunk 8]
```diff
@@ -4,12 +4,24 @@
...
 import javafx.scene.control.Label;
 import javafx.stage.Stage;
+import services.ServiceLocator;
 
 /**
  * Factory for JavaFX constructors.
  */
 public class FXElementsFactory {
 
+    private FXElementsFactory() {
```
Comment (MStraatman) [inline src/main/java/screens/FXElementsFactory.java]: Javadoc.

File: src/main/java/screens/FXElementsFactory.java (hunk 4 above)
Comment (MStraatman) [inline src/main/java/screens/FXElementsFactory.java]: Javadoc.

File: src/main/java/screens/FileSelector.java [hunk 9]
```diff
@@ -30,11 +29,21 @@
      */
     private static Logger logger;
 
+    private static ServiceLocator serviceLocator;
```
Comment (MStraatman) [inline src/main/java/screens/FileSelector.java]: Javadoc.

File: src/main/java/screens/FileSelector.java [hunk 10]
```diff
@@ -30,11 +29,21 @@
...
+
     /**
      * Private constructor.
      */
     private FileSelector() { }
 
+    public static void register(ServiceLocator sL) {
```
Comment (MStraatman) [inline src/main/java/screens/FileSelector.java]: Javadoc.

File: src/main/java/screens/scenes/InteractionScene.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
+package screens.scenes;
+
+import javafx.scene.layout.VBox;
+import services.ServiceLocator;
+
+/**
+ * Created by Michael on 6/8/2017.
```
Comment (tvheems) [inline src/main/java/screens/scenes/InteractionScene.java]: Please change to "101010".

File: src/main/java/services/ServiceLocator.java [hunk 12]
```diff
@@ -0,0 +1,139 @@
+package services;
```
Comment (tvheems) [inline src/main/java/services/ServiceLocator.java]: A lot of javadoc is missing in this class, please update it.

File: src/main/java/screens/nodehandlers/INodeHandler.java [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+     */
+    void handleNode(DrawNode node);
+
+    /**
+     * Executes operations when a line is selected.
+     * @param line Line object to work with.
+     */
+    void handleLine(Line line);
+
+}
```
Comment (MStraatman) [inline src/main/java/screens/nodehandlers/INodeHandler.java]: No empty line at end of the file.

File: src/main/java/screens/nodehandlers/NodeCenter.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleNode(DrawNode node) {
+        serviceLocator.getGraphScene().drawGraph(node.getIndex(), serviceLocator.getController().getRadius());
+        serviceLocator.getGraphScene().switchToInfo();
+    }
+
+    @Override
```
Comment (MStraatman) [inline src/main/java/screens/nodehandlers/NodeCenter.java]: Javadoc.

File: src/main/java/screens/nodehandlers/NodeInfo.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleNode(DrawNode node) {
+        serviceLocator.getInfoScreen().displayNodeInfo(node);
+    }
+
+    @Override
```
Comment (MStraatman) [inline src/main/java/screens/nodehandlers/NodeInfo.java]: Javadoc.

File: src/main/java/screens/scenes/GraphScene.java [hunk 16]
```diff
@@ -43,30 +48,37 @@
      */
     private FXElementsFactory fxElementsFactory;
 
+    private ServiceLocator serviceLocator;
```
Comment (MStraatman) [inline src/main/java/screens/scenes/GraphScene.java]: Javadoc.

File: src/main/java/screens/scenes/GraphScene.java [hunk 17]
```diff
@@ -43,30 +48,37 @@
...
+     public GraphScene(ServiceLocator sL) {
+         center = new NodeCenter(sL);
+         info = new NodeInfo(sL);
          state = info;
-         this.fxElementsFactory = fact;
+         this.fxElementsFactory = sL.getFxElementsFactory();
+         this.serviceLocator = sL;
+     }
+
+     public static void register(ServiceLocator sL) {
```
Comment (MStraatman) [inline src/main/java/screens/scenes/GraphScene.java]: Javadoc

File: src/main/java/services/ServiceLocator.java [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+import screens.Backlog;
+import screens.FXElementsFactory;
+import screens.FileSelector;
+import screens.scenes.Controller;
+import screens.scenes.GraphScene;
+import screens.scenes.InfoScreen;
+import screens.scenes.InteractionScene;
+
+/**
+ * Created by Michael on 6/8/2017.
```
Comment (MStraatman) [inline src/main/java/services/ServiceLocator.java]: Please change this.

File: src/main/java/services/ServiceLocator.java [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+ */
+public class ServiceLocator {
```
Comment (MStraatman) [inline src/main/java/services/ServiceLocator.java]: Javadocs for everything in class is still needed.

File: target/classes/layoutstyles.css [hunk 20]
```diff
@@ -3,8 +3,25 @@
...
+    -fx-alignment: top-center;
+    -fx-padding: 10;
+    -fx-spacing: 5;
+    -fx-hgap: 10;
+    -fx-vgap: 10;
+}
+
+#title {
+    -fx-alignment: top-left;
 }
```
Comment (MStraatman) [inline target/classes/layoutstyles.css]: There should be an empty line here.

File: src/test/java/screens/scenes/GraphSceneTest.java [hunk 21]
```diff
@@ -15,14 +15,20 @@
 import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
+import screens.FXElementsFactory;
+import services.ServiceLocator;
 
 import javax.swing.*;
```
Comment (MStraatman) [inline src/test/java/screens/scenes/GraphSceneTest.java]: This *-import can be changed to: javax.swing.SwingUtilities. Please do so.

File: src/main/resources/layoutstyles.css [hunk 22]
```diff
@@ -3,8 +3,25 @@
...
+    -fx-alignment: top-center;
+    -fx-padding: 10;
+    -fx-spacing: 5;
+    -fx-hgap: 10;
+    -fx-vgap: 10;
+}
+
+#title {
+    -fx-alignment: top-left;
 }
```
Comment (MStraatman) [inline src/main/resources/layoutstyles.css]: Should be an empty line.

File: src/main/java/screens/scenes/InteractionScene.java [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+     * @param serviceLocator ServiceLocator for locating services registered in that object.
+     */
+    public InteractionScene(ServiceLocator serviceLocator) {
+        infoScreen = new InfoScreen(serviceLocator);
+        controller = new Controller(serviceLocator);
+        setSettings();
+    }
+
+    public static void register(ServiceLocator sL) {
+        if(sL == null) {
```
Comment (MStraatman) [inline src/main/java/screens/scenes/InteractionScene.java]: Whitespace after if-statement.

File: src/main/java/screens/FXElementsFactory.java [hunk 24]
```diff
@@ -4,12 +4,24 @@
...
+
+    }
+
+    public static void register(ServiceLocator sL) {
+        if(sL == null) {
```
Comment (MStraatman) [inline src/main/java/screens/FXElementsFactory.java]: Whitespace after if-statement, please add it for all register methods.

**Review by tvheems — CHANGES_REQUESTED** [review]
The GUI looks great, but tomato takes a long time to parse, maybe a performance issue somewhere?
A lot of javadoc is missing.
Also, run your checkstyle, there are some unresolved warnings.

**Review by MStraatman — CHANGES_REQUESTED** [review]
I like the new GUI and also how the functionality now gets split better, because before this, a lot of functionality was just in the main Window class. There are a lot of checkstyle errors I've pointed out, that I want to see fixed, before I can approve this.

**Review by MStraatman — APPROVED** [review]

**Comment by tvheems** [comment]
There are still some minor check style errors, please fix these before merging.

**Review by tvheems — APPROVED** [review]

**Pull request merged**